GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -434,18 +445,18 @@ private Integer parseRepeatsForInstance(InstanceData instanceData,
         // 8. SurveyalTime - ignored duplicate
         // 9 - N. Questions
 
-        int instanceIdColumnIndex = metadataColumnHeaderIndex2.get(SURVEY_INSTANCE_COLUMN_KEY);
-        int repeatIterationColumnIndex = metadataColumnHeaderIndex2.get(REPEAT_COLUMN_KEY);
+        int identifierColumnIndex = metadataColumnHeaderIdx.get(DATAPOINT_IDENTIFIER_COLUMN_KEY);
+        int repeatIterationColumnIndex = metadataColumnHeaderIdx.get(REPEAT_COLUMN_KEY);
 
-        String SurveyInstanceId = instanceData.surveyInstanceDto.getKeyId().toString();
+        String dataPointIdentifier = instanceData.surveyInstanceDto.getSurveyedLocaleIdentifier();
         //Find first row matching this, or give up
         int rowIx = currentRowIndex;
         Row row;
         while (true) {
             row = repSheet.getRow(rowIx);
             if (row != null
-                    && row.getCell(instanceIdColumnIndex) != null
-                    && row.getCell(instanceIdColumnIndex).getStringCellValue().equals(SurveyInstanceId)) { //found!
+                    && !isEmptyCell(row.getCell(identifierColumnIndex))
+                    && row.getCell(identifierColumnIndex).getStringCellValue().equals(dataPointIdentifier)) { //found!
                 break;
             } else {
                 if (isEmptyRow(row)) { // a row without any cells defined
```

Comment:
I was actually referring to this line i.e. line 462.  since we are checking for a value in the datapoint identifier column why scan the entire row?